_books/ion-1-1/src/text/values.md

```diff
@@ -0,0 +1,489 @@
+
+```ion
+// A valid blob value with zero padding characters.
+{{
+  +AB/
+}}
+
+// A valid blob value with one required padding character.
+{{ VG8gaW5maW5pdHkuLi4gYW5kIGJleW9uZCE= }}
+
+// ERROR: Incorrect number of padding characters.
+{{ VG8gaW5maW5pdHkuLi4gYW5kIGJleW9uZCE== }}
+
+// ERROR: Padding character within the data.
+{{ VG8gaW5maW5pdHku=Li4gYW5kIGJleW9uZCE= }}
+
+// A valid blob value with two required padding characters.
+{{ dHdvIHBhZGRpbmcgY2hhcmFjdGVycw== }}
+
+// ERROR: Invalid character within the data.
+{{ dHdvIHBhZGRpbmc_gY2hhcmFjdGVycw= }}
+```
+
+## Lists
+
+In the text format, lists are bounded by square brackets and elements
+are separated by commas.
+
+```ion
+[]                // An empty list value
+[1, 2, 3]         // List of three ints
+[ 1 , two ]       // List of an int and a symbol
+[a , [b]]         // Nested list
+[ 1.2, ]          // Trailing comma is legal in Ion (unlike JSON)
+[ 1, , 2 ]        // ERROR: missing element between commas
+```
+
+## S-expressions
+
+In the text format, S-expressions are bounded by parentheses.
+S-expressions also allow unquoted operator symbols, in addition to the
+unquoted identifier symbols allowed everywhere.
+
+```ion
+()                // An empty expression value
+(cons 1 2)        // S-expression of three values
+([hello][there])  // S-expression containing two lists
+
+(a+-b)  ( 'a' '+-' 'b' )    // Equivalent; three symbols
+(a.b;)  ( 'a' '.' 'b' ';')  // Equivalent; four symbols
+```
+
+Note that comments are allowed within S-expressions and have higher precedence
+than operators, therefore `//` and `/*` denote the start of comment blocks.
+Users are advised to avoid them as operators, though they can be used when
+escaped with single quotes:
+```ion
+(a/* word */b)       // An S-expression with two symbols and a comment
+(a '/*' word '*/' b) // An S-expression with five symbols
+```
+
+## Structs
+
+In the text format, a `struct` is wrapped by curly braces, with a colon between each name and value, and a comma between the fields.
+The field name is a [symbol token](symbol-tokens.md).
+For the purposes of JSON compatibility, it is also legal to use a string for field names, but they are converted to symbol tokens by the parser.
+
+```ion
+{ }                                 // An empty struct value
+{ first : "Tom" , last: "Riddle" }  // Structure with two fields
+{"first":"Tom","last":"Riddle"}     // The same value with confusing style
+{center:{x:1.0, y:12.5}, radius:3}  // Nested struct
+{ x:1, }                            // Trailing comma is legal in Ion (unlike JSON)
+{ "":42 }                           // A struct value containing a field with an empty name
+{ x:1, x:null.int }                 // WARNING: repeated name 'x' leads to undefined behavior
+{ x:1, , }                          // ERROR: missing field between commas
+```
+
+Note that field names are symbol *tokens*, not symbol *values*, and thus may not be annotated.
```

Comment:
There should probably be some mention of using an e-expression in field name position.

Reply:
I will add it to the e-expression page.